File: mpq_test.go
package main

import (
	"os"
	"path/filepath"
	"testing"
)

func TestSniffRecognizesUARFixtures(t *testing.T) {
	for _, name := range []string{"20260723-1802.SC2Replay", "20260723-1808.SC2Replay"} {
		data, err := os.ReadFile(filepath.Join("testdata", name))
		if err != nil {
			t.Fatal(err)
		}
		uar, err := isUARReplay(data)
		if err != nil {
			t.Fatalf("%s: %v", name, err)
		}
		if !uar {
			t.Errorf("%s: expected UAR title to be found", name)
		}
	}
}

func TestSniffRejectsJunk(t *testing.T) {
	if _, err := isUARReplay([]byte("this is definitely not an MPQ archive, not even close")); err == nil {
		t.Error("junk bytes should not parse as a replay")
	}
	if _, err := isUARReplay([]byte{}); err == nil {
		t.Error("empty input should not parse as a replay")
	}
	// truncated real replay: valid magic, tables point past the end
	data, err := os.ReadFile(filepath.Join("testdata", "20260723-1808.SC2Replay"))
	if err != nil {
		t.Fatal(err)
	}
	if _, err := isUARReplay(data[:200]); err == nil {
		t.Error("truncated replay should error, not crash")
	}
}

File: tests/mpq.test.ts
import { test } from 'node:test';
import assert from 'node:assert/strict';
import { readFileSync } from 'node:fs';
import { isUARReplay } from '../src/core/sniff.ts';

test('sniff recognizes UAR fixtures', () => {
	for (const name of ['20260723-1802.SC2Replay', '20260723-1808.SC2Replay']) {
		const data = readFileSync(new URL(`../testdata/${name}`, import.meta.url));
		assert.equal(isUARReplay(data), true, `${name}: expected UAR title to be found`);
	}
});

test('sniff rejects junk', () => {
	assert.throws(() => isUARReplay(Buffer.from('this is definitely not an MPQ archive')));
	assert.throws(() => isUARReplay(Buffer.alloc(0)));
	// truncated real replay: valid magic, tables point past the end
	const data = readFileSync(new URL('../testdata/20260723-1808.SC2Replay', import.meta.url));
	assert.throws(() => isUARReplay(data.subarray(0, 200)));
});
